### PR TITLE
Don't filter results when opening a codecompletion session

### PR DESCRIPTION
**Pull request by ahoppen**
Filtering the results in SourceKit takes considerable time, which we are not really interested in measuring.

Retrieve the initial set of results without a filter text with a result limit of 200. If we need to check for an expected result, perform a subsequent code completion update that sets the filter text. Don't measure the performance of that request.

### CODE REVIEW

File: SourceKitStressTester/Sources/StressTester/SourceKitDocument.swift [hunk 1]
```diff
@@ -232,60 +232,71 @@ class SourceKitDocument {
...
+    request.addCompilerArgs(args.sourcekitdArgs)
+
+    let completionOptions = request.addDictionaryParameter(.key_CodeCompleteOptions)
+    // Don't hide any results because otherwise checking if the expected result is contained will fail
+    completionOptions.add(.key_HideLowPriority, value: 0)
+    completionOptions.add(.key_HideByName, value: 0)
+    completionOptions.add(.key_HideUnderscores, value: 0)
+
+    // Set expected results to 200 to avoid inflating the time measurements by serialization time.
+    // To make sure that the expected result is in the results, filter by its base name.
```
Comment (bnbarham) [inline SourceKitStressTester/Sources/StressTester/SourceKitDocument.swift:258]: Comment is out of date

File: SourceKitStressTester/Sources/StressTester/SourceKitDocument.swift [hunk 2]
```diff
@@ -232,60 +232,71 @@ class SourceKitDocument {
...
+  func codeComplete(offset: Int, expectedResult: ExpectedResult?) throws -> (request: RequestInfo, response: SourceKitdResponse, instructions: Int, reusingASTContext: Bool) {
+    var responseToReport: SourceKitdResponse
+    let openRequest = buildCompletionRequest(request: .request_CodeCompleteOpen, offset: offset, extraOptions: extraCodeCompleteOptions)
+    let openInfo = RequestInfo.codeCompleteOpen(document: documentInfo, offset: offset,
+                                                args: args.sourcekitdArgs)
+    let (openResponse, openInstructions) = try sendWithTimeoutMeasuringInstructions(openRequest, info: openInfo)
+
+    let reusingASTContext = openResponse.value.getBool(.key_ReusingASTContext)
+
+    if let expectedResult = expectedResult {
```
Comment (bnbarham) [inline SourceKitStressTester/Sources/StressTester/SourceKitDocument.swift:280]: Will be interesting to see how much slower the extra request makes it.

**Review by bnbarham — APPROVED** [review]

File: SourceKitStressTester/Sources/StressTester/SourceKitDocument.swift (hunk 2 above)
Comment (ahoppen) [reply SourceKitStressTester/Sources/StressTester/SourceKitDocument.swift:280]: I would expect that the update should be pretty fast, but let’s see. I started a stress tester run via https://github.com/apple/swift/pull/37710.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
Doesn’t seem to have made a measurable runtime difference for the stress tester. Execution of https://ci.swift.org/job/swift-PR-macos-with-sourcekit-stress-tester/78/ finished in 10 hours, which matches previous runs.

**Pull request merged**